### ISSUE TITLE
Swig + scilab creates Init function that conflicts with internal function name

### ISSUE THREAD

**Issue by accesio** [swig/swig#745]
Hi,

If you have a C file called foo.c with the function name

Foo_Init() {

}

And then you have a SWIG file defined as follows:

%module Foo

You will get an error because Scilab Swig will create a function called Foo_Init() and this causes a compilation error.

[example.zip](https://github.com/swig/swig/files/359534/example.zip)

**Comment by smarchetto** [comment]
fixed with [d0a45be](https://github.com/swig/swig/commit/d0a45be1ebfe125d51696b48cb5d359bc7431052)

**Comment by accesio** [comment]
Hi,

This bug fix doesn't appear to have fixed the problem.

Using d0a45be  release of swig and my attached example I get:

```bash
gcc -fPIC -shared -o libFoo.so  -I.  Foo_wrap.c foo.c -I/usr/include/scilab/

Foo_wrap.c:1507:5: error: conflicting types for ‘Foo_Init’
 int Foo_Init(SWIG_GatewayParameters) {
     ^
In file included from Foo_wrap.c:1223:0:
foo.h:4:6: note: previous declaration of ‘Foo_Init’ was here
 void Foo_Init();
      ^
```

**Comment by ojwb** [comment]
@smarchetto I confirmed this isn't fixed with current git master (I don't have the scilab stuff needed to compile the example, but inspecting the code I can see there are conflicting `Foo_Init()` declarations - here's a full reproducer:

{{{
%module Foo
%inline %{
Foo_Init() { }
%}
}}}

Process with: `swig -scilab foo.i`

I don't know anything about scilab, but a quick look at the generated wrapper suggests to me you just need to prefix the name of the generated `Foo_Init` C function with `SWIG_`.

**Comment by accesio** [comment]
Hey,

I attached a zip file that reproduces this problem exactly.

Just compile with the gcc statement I put up above.

**Comment by smarchetto** [comment]
Sorry my comment was a mistake, the fix closed the other issue [#746] (https://github.com/swig/swig/issues/746)
